client/question/components/SpeechBalloonDirectives.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for displaying speech balloons.
+ */
+
+tie.directive('tieSpeechBalloonContainer', ['$timeout', function($timeout) {
+  return {
+    restrict: 'A',
+    link: function(scope, iElement) {
+      if (iElement[0].nodeName !== "DIV") {
+        return;
+      }
+      var speechBalloonContainer = iElement[0];
+      speechBalloonContainer.className = "tie-speech-balloon-container";
+      speechBalloonContainer.style.opacity = "0";
+      speechBalloonContainer.style.transition = "unset";
+      speechBalloonContainer.style.display = "none";
+      $timeout(function() {
+        speechBalloonContainer.style.display = "block";
+        speechBalloonContainer.style.marginTop =
+            '-' + speechBalloonContainer.offsetHeight.toString() + 'px';
+        $timeout(function() {
+          speechBalloonContainer.removeAttribute("style");
+        }, 0);
+      }, 0);
+    }
+  };
+}]);
+
+tie.directive('tieSpeechBalloonLeft', ['$timeout', function($timeout) {
```

Comment:
Prefer having one directive per file. OK to put all the files in a subfolder if desired.

Reply:
I thought about that, but these are all very tightly related and used together (doesn't really make sense to use any one in isolation). If you feel strongly about this, I will split them up into different files.

Reply:
Yeah, I'm quite torn here for the reason you mention. Let's leave it as is for now. These feel more like animation tags (and not so much "DOM structure") and I'll need to think of a more general way to handle these / distinguish these from other more "structural" directives.